lib/WebpackOptionsDefaulter.js

```diff
@@ -432,7 +443,14 @@ class WebpackOptionsDefaulter extends OptionsDefaulter {
 						sourceMap:
 							(options.devtool && /source-?map/.test(options.devtool)) ||
 							(options.plugins &&
-								options.plugins.some(p => p instanceof SourceMapDevToolPlugin))
+								options.plugins.some(p => p instanceof SourceMapDevToolPlugin)),
+						terserOptions: {
+							module: options.output.module,
+							ecma:
+								options.output.ecmaVersion > 2000
+									? options.output.ecmaVersion - 2009
```

Comment:
`options.output.ecmaVersion: 2009` doesn't work as expected and generate invalid `ecma` version. I think will be great disable usage `20*` option for `ecmaVersion`:
- better schema validation
- some developers can do mistake like `2029` or `20029`
- new version of ecma may not be released every year

Reply:
The schema should handle this

https://github.com/webpack/webpack/blob/08dd5adf81f48207bbe8cd523a0e3af493493f57/schemas/WebpackOptions.json#L1158-L1169

The official ecma version naming is by year. it's ES3 -> ES5 -> ES2015 -> ES2016 -> ...

> * some developers can do mistake like `2029` or `20029`

we could add a maximum of i. e. 3000 to catch 20029.
2029 is a valid (not yet released) ecma version, so this seem fine to me.